### ISSUE TITLE
Sharing Service: Invalid Schema Version

### ISSUE THREAD

**Issue by DrNeuroSurg** [Microsoft/MixedRealityToolkit-Unity#487]
If using SharingService with schema version 15 on client and server everything works fine (as in HoloToolkit-Sharing-240).

With schema version 17 it isn't working anymore !

**Comment by HodgsonSDAS** [comment]
Most likely related to the major sharing update (https://github.com/Microsoft/HoloToolkit-Unity/pull/459) that introduced multiple problems.
Thanks for submitting.

**Comment by Lordinaire** [comment]
I confirm, my application crashes when I try to use:
`SharingManager.Create(config);`

**Comment by HodgsonSDAS** [comment]
If anyone has a fix, I've already submitted PR https://github.com/Microsoft/HoloToolkit-Unity/pull/484 for other misc fixes for the sharing service.

Please submit a PR on my branch to get those changes included to the existing PR.

**Comment by HodgsonSDAS** [comment]
@Lordinaire are you using the SharingStage.cs class or do you have your own implementation?

I have not been able to get my application to crash on `SharinManager.Create(config);`

**Comment by HodgsonSDAS** [comment]
**RESOLUTION**
In the main HTK project launching the SharingService.exe always launches with schema v17.

If you're using the HoloToolkit from the unity package, or exported the package yourself chances are you  got your SharingService.exe from another source. You just need to replace that old executable with the newest version.

For example, in my project, I use a symbolic link to reference the HTK specific libraries in a folder named ThirdParty, and my External\HoloToolkit\Sharing folder doesn't get updated when the HTK updates. (I could just sym link those too, but I'm lazy).

**Comment by NeerajW** [comment]
Great catch @HodgsonSDAS !!

**Comment by Lordinaire** [comment]
@HodgsonSDAS I'm using the `SharingStage.cs` class and the latest version of HTK. I create a very simple project with just basic sharing scripts (see attached file) and it crashes after few seconds on my HoloLens. No exception are thrown :/

PS : I removed the HTK to be able to upload the project.

[SharedHologram.zip](https://github.com/Microsoft/HoloToolkit-Unity/files/744630/SharedHologram.zip)

**Comment by HodgsonSDAS** [comment]
Sorry @Lordinaire I'm not going to download your zip file.  You can share on github if you'd like.

**Comment by Lordinaire** [comment]
Of course ! I put it on GitHub : [https://github.com/Lordinaire/HoloLens_Sharing_Test](https://github.com/Lordinaire/HoloLens_Sharing_Test)
If anyone can try to repro the crash it will help me a lot.

**Comment by HodgsonSDAS** [comment]
Have you tried pulling my PR addressing some of the sharing services?

**Comment by GJ0hnson** [comment]
Hello,
I am working on a project using the sharing components of the HoloToolKit for Unity. I am using the current version of the HoloToolKit in a Unity and still receiving issues when launching the sharingservice.exe, similar to @Lordinaire. The sharingservice.exe does say Schema Version: 15 when launched, but as soon as I attempt to connect to the server address I receive the Error code 5, exactly as pictured above in the post from @StephenHodgson. I made sure to check the resolution previously posted, but I have already completed those steps and still have issues. Additionally, I was able to get the sharing services to work for the HoloToolKit-240, but am having issues here. Any updated resolutions or advice would be greatly appreciated!
Thank you in advance.

**Comment by Siman007** [comment]
Hi have a problem with the HoloToolKit for Unity. I am using the current version of the HoloToolKit in a Unity and still receiving issues when launching the sharingservice.exe. When I add the ImportExportAnchorManager the app dies as it loads on hololens with no debug. Tracing it through it appears to be after line 225 in ImportExportAnchorManager which in my build reads  SharingStage.Instance.SharingManagerConnected += Connected;

private void Start()
        {
            // SharingStage should be valid at this point, but we may not be connected.
            if (SharingStage.Instance.IsConnected)
            {
                Connected();
            }
            else
            {
               SharingStage.Instance.SharingManagerConnected += Connected;
            }
        }

The only debug I get after this line is in SharingStage.cs in HandleLog which is an Assert with the log message logString = "WinRTScriptingObjectWrapper::InternalAddRef Reference Count cannot be zero"

Any ideas gratefully received

**Comment by StephenHodgson** [comment]
@Siman007, I suggest opening a new issue instead of posting off topic. Thanks

**Comment by RaoAM** [comment]
HI, I have a problem with sharing service. I developed few hololens apps using HTK V 1.5.6. For all those apps I used sharing service executable file provided in HTK 1.5.6 with schema 15  hosted on a public IP. 

Recently I  developed 2 hololens apps with VOIP features using HTK V 1.5.8. In order to make this app work, I updated my sharing server with the new version available in 1.5.8 with schema 17. After that all my old apps using sharing service stop working. 
With out migrating my previous apps, how can I make the sharing server to understand schema 17 and previous schema 14.

Thank you in advance
